### PR TITLE
feat: auth contract

### PR DESCRIPTION
**Pull request by charleenfei**

### CODE REVIEW

File: src/auth.sol [hunk 1]
```diff
@@ -0,0 +1,25 @@
+// Copyright (C) Centrifuge 2020
```
Comment (xmxanuel) [inline src/auth.sol]: 2020 💛

Comment (xmxanuel) [reply src/auth.sol]: lets add a reference to maker

File: src/auth.sol [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+// GNU Affero General Public License for more details.
+//
+// You should have received a copy of the GNU Affero General Public License
+// along with this program.  If not, see <https://www.gnu.org/licenses/>.
+
+pragma solidity >=0.5.12;
+
+import "ds-note/note.sol";
+
+contract Auth is DSNote {
```
Comment (xmxanuel) [inline src/auth.sol:20]: should we have `DSNote` here if we are not using it?

**Review by xmxanuel — APPROVED** [review]
LGTM.

a comment about DSNote

File: src/auth.sol (hunk 2 above)
Comment (charleenfei) [reply src/auth.sol:20]: we use it in rely and deny

**Pull request merged**